scripts/criu-ns

```diff
@@ -61,39 +61,87 @@ def run_criu():
 	os.execlp('criu', *['criu'] + sys.argv[1:])
 
 
+def _mount_new_proc():
+	"""
+	Mount new /proc filesystem.
+	"""
+	if _mount(None, b"/", None, MS_SLAVE|MS_REC, None):
+		_errno = ctypes.get_errno()
+		raise OSError(_errno, errno.errorcode[_errno])
+	if _mount(b'proc', b'/proc', b'proc', 0, None):
+		_errno = ctypes.get_errno()
+		raise OSError(_errno, errno.errorcode[_errno])
+
+
+def _wait_for_process_status(criu_pid):
+	"""
+	Wait for CRIU to exit and report the status back.
+	"""
+	while True:
+		try:
+			(pid, status) = os.wait()
+			if pid == criu_pid:
+				return os.WEXITSTATUS(status)
+		except OSError:
+			return -251
+
+
+def wrap_restore_detached():
+	"""
+	Create an init process to keep the new PID namespace
+	after CRIU exits.
+	"""
+	(r_pipe, w_pipe) = os.pipe()
+	# Spawn the init
+	if os.fork() == 0:
+		os.close(r_pipe)
+		_mount_new_proc()
+		# Spawn CRIU binary
+		criu_pid = os.fork()
```

Comment:
Hmmm. I think you are going to reintroduce the bug that you fixed a month ago:
https://github.com/checkpoint-restore/criu/commit/265569a40f04326dc2b310eb5ddbac9c1cb989e3

Reply:
As far as understand when using `--restore-detached`, CRIU exits after restore, therefore we need another process to keep running within the namespace as PID=1 after restore. As illustrated in https://criu.org/Tree_after_restore#Detached

When CRIU is used without `--restore-detached`, it will run as PID=1.

Reply:
Probably https://github.com/checkpoint-restore/criu/commit/265569a40f04326dc2b310eb5ddbac9c1cb989e3 could've been fixed with just forking criu with some unused pid which will not intersect with restored pids, and this way we would've hit two birds with one stone.

Reply:
@rst0git could you explain how you resolved this comment?

Reply:
As reported in https://github.com/checkpoint-restore/criu/issues/1278, `-d` when used with `criu-ns` is currently broken. It does not work because CRIU exists after restore to detach itself from the process tree, and if  CRIU runs as PID=1 all processes in the PID namespace will be terminated.
This issue was introduced with https://github.com/checkpoint-restore/criu/commit/265569a40f04326dc2b310eb5ddbac9c1cb989e3. 

> could you explain how you resolved this comment?

To fix the broken `-d` functionality, I'm reusing the previous code/behaviour before the commit mention above. Although it has the limitation mentioned in this comment, it works.

In other words, I haven't (yet) resolved the problem mentioned in this comment. It remains a limitation of `criu-ns`.